testsuite/features/support/twopence_init.rb

```diff
@@ -335,11 +335,15 @@ def client_public_ip(host)
   next if node.nil?
   next if node.is_a?(String) && node.empty?
 
-  node.init_ip(node.full_hostname)
-
   host = $host_by_node[node]
   raise "Cannot resolve host for node: '#{node.hostname}'" if host.nil? || host == ''
 
+  if ADDRESSES.has_key? host
+    node.init_ip(net_prefix + ADDRESSES[host])
+  else
+    node.init_ip(node.full_hostname)
```

Comment:
why the f*** were we doing that? ;-)

Reply:
So, we want to have in "ip" attribute an IP not a fqdn. Before this change, we were doing `node.init_ip(node.full_hostname)` but I want "ip" attribute of node to contain the private IP if that exist. As it only exists if it's contained in the hash ADDRESSES, I first check it, and if we have private IP I set it, if not I let as it was, with fqdn.

Keep in mind we also have, public_ip attribute, which contains the public ip in all the cases.

Reply:
I think I remember Dario doing that long time ago, and I was telling him "don't call `ip` a variable that holds a FQDN"...